### PR TITLE
Add support for `REGCLASSOID` type

### PR DESCRIPTION
**Pull request by Y--**
Small QoL PR that does two minor thing:
* shows the Oid when the type in unsupported
* add mapping for PG's `REGCLASSOID` type

### CODE REVIEW

**Comment by wuputah** [comment]
Could you please add a test case for this type? Once that's done, lgtm!

**Review by wuputah — APPROVED** [review]

**Review by wuputah — APPROVED** [review]

File: src/pgduckdb_types.cpp [hunk 1]
```diff
@@ -543,8 +543,12 @@ ConvertPostgresToDuckColumnType(Form_pg_attribute &attribute) {
 		}
 		return duck_type;
 	}
-	default:
-		return duckdb::LogicalType::USER("UnsupportedPostgresType");
+	case REGCLASSOID:
+		return duckdb::LogicalTypeId::INTEGER;
```
Comment (Tishj) [inline src/pgduckdb_types.cpp:547]: That works, do keep in mind that this means we lose the context of what this was on the postgres side this way.
We'd need something similar to what we do for `NumericAsDouble` to differentiate between an actual INT4 and a REGCLASSOID

**Comment by hlinnaka** [comment]
OIDs are unsigned, so this goes wrong:
```
postgres=# create table x (cls regclass);
CREATE TABLE
postgres=# insert into x values (3_000_000_000);
INSERT 0 1
postgres=# select * from x;
    cls     
------------
 3000000000
(1 row)

postgres=# SET duckdb.execution TO true;
SET
postgres=# select * from x;
     cls     
-------------
 -1294967296
(1 row)

```

**Comment by Y--** [comment]
> OIDs are unsigned, so this goes wrong:
> 
> ```
> postgres=# create table x (cls regclass);
> CREATE TABLE
> postgres=# insert into x values (3_000_000_000);
> INSERT 0 1
> postgres=# select * from x;
>     cls     
> ------------
>  3000000000
> (1 row)
> 
> postgres=# SET duckdb.execution TO true;
> SET
> postgres=# select * from x;
>      cls     
> -------------
>  -1294967296
> (1 row)
> ```

Thanks for catching this! I'll work on this ASAP

**Comment by Y--** [comment]
Will be fixed here: https://github.com/duckdb/pg_duckdb/pull/125

**Pull request merged**